pkg/terraform/store.go

```diff
@@ -57,6 +55,48 @@ type ProviderRequirement struct {
 // ProviderConfiguration holds the setup configuration body
 type ProviderConfiguration map[string]any
 
+// ToProviderHandle converts a provider configuration to a handle
+// for the provider scheduler.
+func (pc ProviderConfiguration) ToProviderHandle() (ProviderHandle, error) {
```

Comment:
I think we are using the ProviderConfig hash, for deciding whether we need a new tf-provider process to, right? By this way, we will resolve the resource leak issues that we observed before.

Reply:
Yes, correct. `ProviderHandle` is basically a hash of the Terraform provider configuration block. Clients pass a `ProviderHandle` to the scheduler while making reservation requests to identify their requests.